metaflow/plugins/aws/batch/batch.py

```diff
@@ -262,6 +262,7 @@ def create_job(
             .environment_variable("METAFLOW_DEFAULT_DATASTORE", "s3")
             .environment_variable("METAFLOW_DEFAULT_METADATA", DEFAULT_METADATA)
             .environment_variable("METAFLOW_CARD_S3ROOT", CARD_S3ROOT)
+            .environment_variable("METAFLOW_S3_WORKER_COUNT", max(1, int(cpu) - 2))
```

Comment:
would we need a change in step functions as well?

Reply:
step-functions calls the batch implementation so this should apply there as well. will verify

Reply:
confirmed, step-functions is setting the env correctly through the changes in Batch